### PR TITLE
chore: enforce compact blank-line formatting

### PR DESCRIPTION
**Pull request by DysektAI**
## Summary

- Move the scoped blank-line cleanup into a dedicated worktree branch.
- Remove unnecessary blank lines from supported project source and configuration files.
- Add a parser-aware blank-line checker and wire it into lint, formatting, and lint-staged workflows.
- Preserve intentional blank lines in Markdown, embedded template strings, generated files, and multiline configuration blocks.

## Why

The existing ESLint rule only covered the `app` directory, so scripts, workflows, Supabase functions, workers, and other project configuration could still accumulate unnecessary blank lines. This adds repository-wide enforcement while avoiding content where blank lines are meaningful.

## Validation

- `pnpm run lint`
- `pnpm run format:check`
- `pnpm run typecheck`
- `pnpm --filter api-gateway run typecheck`
- `pnpm run test:api-gateway`
- `pnpm run test` — 219 test files and 2,167 tests passed
- `git diff --check`

<a href="https://cubic.dev/pr/tarkovtracker-org/TarkovTracker/pull/611?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added scheduled or manually triggered precomputation of game data.
  * Added automatic labeling for application-related changes.
* **Developer Experience**
  * Added blank-line formatting and validation to formatting and lint workflows.
  * Expanded staged-file checks across configuration and source files.
* **Documentation**
  * Clarified formatting, validation, workflow automation, and review guidance.
* **Maintenance**
  * Standardized whitespace across project configuration, scripts, workflows, services, and tests.
  * Improved workflow checkout security by disabling persisted credentials.

___

## **CodeAnt-AI Description**
Enforce compact blank-line formatting across supported project files

### What Changed
- Added repository-wide blank-line checking and automatic cleanup for source, configuration, scripts, workflows, and tests.
- Preserves intentional blank lines in Markdown, generated files, Vue template content, shell heredocs, YAML multiline values, comments, and embedded strings.
- Integrated blank-line checks into lint, formatting, and staged-file workflows, with regression coverage for parser edge cases.
- Updated contributor guidance and CI checkout steps to apply the new formatting process consistently and avoid retaining checkout credentials.

### Impact
`✅ Consistent formatting across project source and configuration`
`✅ Fewer formatting failures in lint and CI checks`
`✅ Preserved readable multiline content`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

<details open><summary><h3>Greptile Summary</h3></summary>

This PR introduces repository-wide blank-line enforcement via a new parser-aware script (`scripts/lint-blank-lines.mjs`), integrating it into lint, format, and lint-staged workflows. It also removes unnecessary blank lines from 63 files across workflows, Supabase functions, workers, and configuration.

- **New `lint-blank-lines.mjs`** (~773 lines): a zero-dependency blank-line stripper with full protection for YAML block scalars, shell heredocs, JS/TS template literals, multi-line comments, and Vue SFCs — backed by an extensive integration test suite (`lint-blank-lines.test.mjs`).
- **CI hardening**: `persist-credentials: false` is added to every `actions/checkout` step across all workflow files, reducing the window for credential leakage via injected steps.
- **lint-staged expansion**: The staged-file hooks now cover scripts, Supabase functions, workers, YAML, and root config files in addition to the existing `app/` rules.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge — all changes are pure formatting cleanup plus a new enforcement script that is thoroughly tested and does not touch runtime logic.

The new lint-blank-lines.mjs handles all the tricky edge cases (YAML block scalars, shell heredocs, template literals, Vue SFCs) and is backed by 40+ integration fixtures. The workflow changes add persist-credentials: false uniformly and remove blank lines — no CI jobs lose required auth. The one notable rough edge (overlapping lint-staged patterns causing double execution for script files) is idempotent and has no user-visible impact.

**Files Needing Attention:** The package.json lint-staged section is the most complex part of the diff — worth a quick scan to confirm pattern coverage is complete and the overlap between !(app)/** and scripts/** entries is intentional.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| scripts/lint-blank-lines.mjs | New 773-line parser-aware blank-line enforcement script; handles YAML block scalars, shell heredocs, JS/TS template literals, multi-line comments, and Vue SFCs. Logic is complex but well-covered by the companion test file. |
| scripts/lint-blank-lines.test.mjs | Comprehensive integration test suite with 40+ fixture cases covering YAML scalars, heredocs, shell continuations, Vue SFCs, JSX, fallback parser mode, and edge cases like empty delimiters and CRLF line endings. |
| scripts/format-files.mjs | Simple Prettier wrapper that replaces the inline glob strings previously in package.json scripts; straightforward and correct. |
| package.json | Scripts updated to chain blank-line checks; lint-staged expanded with many new patterns, but `!(app)/**/*.{js,mjs,cjs}` overlaps with `scripts/**/*.{cjs,js,mjs}`, causing double execution of prettier and blank-line fix for script files. |
| .github/workflows/ci.yml | Blank lines removed from all job steps; `persist-credentials: false` added to every checkout — no functional regressions, security posture improved. |
| .github/workflows/pr-checks.yml | Blank lines removed; persist-credentials disabled on both checkout steps; auto-labeler still receives its own repo-token so no auth regression. |
| .github/workflows/precompute-tarkov-data.yml | Blank lines removed only; persist-credentials was already set to false in the base; no functional changes. |

</details>

<details open><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[staged file] --> B{lint-staged pattern match}
    B -->|app vue/ts/js| C[prettier + eslint + blank-line fix]
    B -->|not-app js/mjs/cjs| D[prettier + blank-line fix]
    B -->|scripts js/mjs/cjs| D2[prettier + blank-line fix AGAIN]
    D --> D2
    B -->|supabase or workers sh/ts/tsx| F[blank-line fix only]
    B -->|yaml or yml| G[blank-line fix only]
    B -->|json files| H[prettier + blank-line fix]
    subgraph CI lint job
    I[pnpm run lint] --> J[ESLint app]
    J --> K[lint:blank-lines run 1]
    end
    subgraph CI format check job
    L[format:check:prettier] --> M[lint:blank-lines run 2]
    end
```
</details>

<sub>Reviews (30): Last reviewed commit: ["fix: preserve staged formatter coverage"](https://github.com/tarkovtracker-org/tarkovtracker/commit/0fe0cb7b864c3aac5600ec2db752d5065aaf52a0) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=49299613)</sub>

**Context used:**

- Context used - AGENTS.md ([source](https://app.greptile.com/dysektai/github/tarkovtracker-org/tarkovtracker/-/custom-context?memory=635c24fb-71e5-43a4-9979-74405b725e95))

### CODE REVIEW

**Comment by DysektAI** [comment]
Addressed the review cycle on commit `1cea2108` and resolved all review threads.

Validation: focused formatter regression tests pass, `lint:blank-lines` passes, `format:check:prettier` passes, commit hooks pass, and the full CI suite is green. The PR remains unmerged and mergeable. Cubic has not published its latest review result yet; its check is the only external item that remained active during polling.

**Comment by DysektAI** [comment]
@cubic-dev-ai review

File: package.json [hunk 1]
```diff
@@ -109,14 +113,23 @@
...
+      "prettier --write",
+      "eslint --fix",
+      "node scripts/lint-blank-lines.mjs --fix"
+    ],
+    "app/**/*.{css,md}": [
+      "prettier --write",
+      "node scripts/lint-blank-lines.mjs --fix"
     ],
     "app/locales/*.json": [
+      "prettier --write",
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply package.json:131]: Fixed in `304f0b6e`. The mutating locale tasks now target only `app/locales/en.json`; the broader `app/locales/*.json` entry runs only the non-mutating i18n validation, so Crowdin-owned exports are not reformatted. I also verified the resolved lint-staged mapping directly.

File: scripts/lint-blank-lines.mjs [hunk 2]
```diff
@@ -0,0 +1,768 @@
...
+      if (/^[\t ]*$/.test(candidate)) {
+        if (keepChomping) {
+          protectedLines.add(next);
+          continue;
+        }
+        const following = lines.slice(next + 1).find((line) => !/^[\t ]*$/.test(line));
+        const followingIndent = following?.match(/^[\t ]*/)?.[0].length;
+        if (contentIndent === null || followingIndent >= contentIndent) protectedLines.add(next);
+        continue;
+      }
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply scripts/lint-blank-lines.mjs:385]: Fixed in `304f0b6e`. All YAML blank predicates now recognize carriage returns, and YAML analysis strips only the trailing CR before parsing while retaining the original CRLF lines for output. The focused regression proves an internal CRLF block-scalar blank line is preserved and the removable trailing blank line is removed.

**Comment by DysektAI** [comment]
Review reconciliation for `304f0b6e`:

- Fixed Prettier failure handling so the wrapper returns the child status without a Node stack trace.
- Fixed excluded-directory matching on path boundaries, with positive coverage for `dist/` and negative coverage for `notdist/`.
- Removed overlapping mutating lint-staged globs after the commit hook demonstrated duplicate execution for scripts; a full tracked-file mapping now reports zero mutating overlaps.
- The latest CodeAnt production-code suggestions target pre-existing behavior unchanged by this formatting-only diff. Valid follow-up work is tracked in #618 (account-deletion concurrency), #619 (team-create rollback), and #620 (token permission validation).

Fresh validation: focused Vitest regression (1/1), `pnpm run lint`, `pnpm run format:check`, `git diff --check`, locale lint-staged mapping assertion, and the Prettier failure-path probe all pass.

File: package.json [hunk 3]
```diff
@@ -109,46 +113,104 @@
...
-    "vitest*.ts": [
+    "./vitest*.ts": [
       "prettier --write",
-      "eslint --fix"
+      "eslint --fix",
+      "node scripts/lint-blank-lines.mjs --fix"
     ],
-    "*.json": [
-      "prettier --write"
+    "./*.{js,mjs,cjs}": [
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply package.json:158]: Fixed in `0fe0cb7b`. Root-only patterns remain explicit, while non-app JS, unknown-directory JSON, and repository-wide Markdown/YAML catchalls preserve coverage without overlapping the directory-specific mutating tasks. I verified the final mapping against every tracked file: zero mutating overlaps and zero uncovered JS/MJS/CJS/JSON/Markdown/YAML files, excluding only Crowdin-owned non-English locales by policy. `pnpm run lint` and `pnpm run format:check` also pass.

**Comment by DysektAI** [comment]
Final review note for `0fe0cb7b`: CodeRabbit reported a successful check status but explicitly marked the latest incremental review as rate limited, so it did not provide fresh review coverage for this head. I performed a direct self-review of the final incremental diff instead: the locale ownership split, CRLF YAML analysis, path-boundary exclusions, child exit propagation, and lint-staged glob coverage/overlap invariants were inspected and exercised. The tracked-file mapping reports zero mutating overlaps and zero uncovered supported JS/JSON/Markdown/YAML files (excluding Crowdin-owned locales by policy); focused Vitest, repository lint, format check, and the complete CI suite pass.
